dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/SecurityOwnershipValidator.java

```diff
@@ -98,15 +96,39 @@ public void validate(Reporter reporter, TrackerBundle bundle, Enrollment enrollm
       }
     }
 
-    checkWriteEnrollmentAccess(reporter, bundle, enrollment, program, ownerOrgUnit, trackedEntity);
+    checkWriteEnrollmentAccess(
+        reporter, bundle, enrollment, program, ownerOrgUnit, trackedEntity.getUid());
+  }
+
+  private TrackedEntity getTrackedEntityWhenStrategyCreate(
+      TrackerBundle bundle, Enrollment enrollment) {
+    TrackedEntity trackedEntity =
+        bundle.getPreheat().getTrackedEntity(enrollment.getTrackedEntity());
+
+    if (trackedEntity == null) {
+      Optional<org.hisp.dhis.tracker.imports.domain.TrackedEntity> te =
+          bundle.findTrackedEntityByUid(enrollment.getTrackedEntity());
+
+      return te.map(
+              entity -> {
+                TrackedEntity newEntity = new TrackedEntity();
+                newEntity.setUid(entity.getUid());
+                newEntity.setOrganisationUnit(
```

Comment:
Why is that mapping necessary?

Reply:
Because the TrackedEntity object returned by findTrackedEntityByUid is not the one the validator expects, so I map it to the expected object.